src/server/status.rs

```diff
@@ -454,26 +451,27 @@ pub fn print_status_all(extended: bool, debug: bool, json: bool)
         );
     }
 
-    let mut tasks = Vec::new();
+    let mut futures = Vec::new();
     let creds_dir = credentials::base_dir()?;
     if let Ok(creds_dir) = fs::read_dir(&creds_dir) {
         for item in creds_dir {
             let item = item?;
             if let Ok(filename) = item.file_name().into_string() {
                 if let Some(name) = filename.strip_suffix(".json") {
                     if !local_names.contains(name) {
-                        tasks.push(task::spawn(
+                        futures.push(
                             RemoteStatus::new(name).probe(item.path())
-                        ));
+                        );
                     }
                 }
             }
         }
     }
+    let mut buffered_futs = stream::iter(futures).buffer_unordered(32);
     let remote_statuses = task::block_on(async {
         let mut rv: Vec<RemoteStatus> = Vec::new();
-        for task in tasks {
-            rv.push(task.await);
+        while let Some(status) = buffered_futs.next().await {
```

Comment:
I think you can do `buffer_unordered(32).collect::<Vec<_>>()`, or `buffer_unordered(32).collect::<Result<Vec<_>>>()`